hist/histv7/src/RAxis.cxx

```diff
@@ -47,6 +47,24 @@ int ROOT::Experimental::RAxisEquidistant::GetBinIndexForLowEdge(double x) const
    return binIdx;
 }
 
+int ROOT::Experimental::RAxisIrregular::GetBinIndexForLowEdge(double x) const noexcept
+{
+   // Check the borders of the bin in which `x` resides
+   const int binIdx = FindBin(x);
+   const double lowBound = GetBinFrom(binIdx);
+   const double upBound = GetBinTo(binIdx);
+
+   // Are we close to one of the bin boundaries?
+   const double tol = 10 * std::numeric_limits<double>::epsilon();
+   if (!IsUnderflowBin(binIdx) && std::fabs(x - lowBound) < tol * std::fabs(lowBound))
```

Comment:
Extract
```
if (IsUnderflowBin(binIdx))
   return -1;
```
and move above `GetBinFrom()` to save some time and to prevent getting undefined bin borders for underflow bins?

Reply:
That would not work, at least not in this form, when x is slightly below the axis minimum, which this function does want to handle. But I guess a reasonable variant is to only query GetBinFrom after the IsUnderflowBin check (and similar for the IsOverflowBin check).

Reply:
Done.